### PR TITLE
JBPM-7418: Stunner - Escalation events

### PR DESCRIPTION
**Pull request by wmedvede**

### CODE REVIEW

**Comment by wmedvede** [comment]
Hello @evacchi @hasys would you mind take a look at this PR?

I hate to put "pressure" but can I ask you to prioritize this review and take a look by today?
This is becuase I won't be here for the next 3 weeks so unfortunately won't be here for the 31 August cut off for the 7.1. :(

Thanks!

**Comment by evacchi** [comment]
I can't tell much about the rest (but it seems fine) the part for new marshallers looks good to me!

**Comment by hasys** [comment]
Jenkins execute full downstream build

**Comment by wmedvede** [comment]
@hasys previous bpmn-icons-sprite had 37 and the new was augmented to have next all the potential icons we'll manage, about 54, so could explain the size change.

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/java/org/kie/workbench/common/stunner/bpmn/definition/property/event/escalation/EscalationRef.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+@Portable
+@Bindable
+@Property
+@FieldDefinition(i18nMode = I18nMode.OVERRIDE_I18N_KEY)
+public class EscalationRef implements BPMNProperty {
+
+    @Value
+    @FieldValue
+    @Pattern(regexp = "^$|[a-zA-Z0-9_]+")
```
[the remark this answers is not in the document]
Comment (wmedvede) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/java/org/kie/workbench/common/stunner/bpmn/definition/property/event/escalation/EscalationRef.java]: Iv'e could this expression only in the EscalationRef and the SignalRef classes.
But even when it's the same regex they represent different things. I don't know to be honest where to place a unique Constant/or whatever with a "unique name" that models this two different things...

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/java/org/kie/workbench/common/stunner/bpmn/definition/property/event/signal/SignalScope.java [hunk 2]
```diff
@@ -52,14 +55,17 @@ public void setValue(final String value) {
 
     @Override
     public int hashCode() {
-        return (null != value) ? value.hashCode() : 0;
+        return Objects.hashCode(value);
```
[the remark this answers is not in the document]
Comment (wmedvede) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/java/org/kie/workbench/common/stunner/bpmn/definition/property/event/signal/SignalScope.java]: @hasys seems like this comment wasn't properly updated but the PR was fixed and re-pushed
small fixes/suggestions here:

https://github.com/kiegroup/kie-wb-common/pull/2030/files#diff-1e55a38cdb61679016772227784470beR61

https://github.com/kiegroup/kie-wb-common/pull/2030/files#diff-1e55a38cdb61679016772227784470beR66

https://github.com/kiegroup/kie-wb-common/pull/2030/files#diff-bc7aa27c8ee18c6a4e76f333857734c6R1892

**Comment by hasys** [comment]
Jenkins execute full downstream build

**Comment by hasys** [comment]
Jenkins retest please

**Comment by hasys** [comment]
Jenkins retest please

**Comment by romartin** [comment]
hey @hasys the builds are ok now. I've tested it and it looks good for me. Please give it a look when you've a chance. Thanks!

**Comment by romartin** [comment]
great, thanks @hasys @LuboTerifaj !

**Pull request merged**